src/android/orb.yml

```diff
@@ -218,4 +218,26 @@ commands:
             [[ -n "$CIRCLE_BRANCH" ]] && PARAMS+=(--branch-name=$CIRCLE_BRANCH)
             [[ -n "$CIRCLE_PULL_REQUEST" ]] && PARAMS+=(--pull-request-url=$CIRCLE_PULL_REQUEST)
             ./gradlew --stacktrace :<<parameters.module_name>>:publishToS3 "${PARAMS[@]}"
+  should-publish-library-artifacts:
```

Comment:
nit: this was unclear to me when I saw this command's name at call site in https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/73/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R44 that the command behind this would be halting the rest of the workflow. Only when I saw the impl below with `step halt` did I get the intent.

What about renaming this to `continue-only-if-publish-library-artefacts` or `check-precondition-for-publish-artefacts`? (the term "precondition" to me kinda hints that it would halt if the precondition fails)

---

Also, I'm not sure of the benefit of having a command for that, as opposed to let the client decide, and make the invocation of that command in a client's job be conditional in the `config.yml` file of the host repo itself? (i.e. https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/73/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R44)

Reply:
I am happy to rename this to `check-precondition-for-publish-artifacts` 👍 

---

> Also, I'm not sure of the benefit of having a command for that, as opposed to let the client decide, and make the invocation of that command in a client's job be conditional in the `config.yml` file of the host repo itself?

@AliSoftware I am sorry, I didn't understand this part. Could you please explain further?

For context, I wanted to use filters instead of a command such as this, but I couldn't find a way to get the behavior we want just using CircleCI filters. @jkmassel also took a look at it and the closest solution we both could find was to enable `only build pull requests` feature in CircleCI which meant that we wouldn't be building `trunk`. He also pointed out, and I agree, that we don't want to enable this option for every library, so it left us to do a command such as this one.

I'd like to get this filter in because some of the libraries (such as gutenberg-mobile) has a very costly setup (~20 min iirc) to get to a stage where we could publish the artifacts.

If you have any suggestions to address these concerns, I'd love to try them, because I am definitely not happy with this solution.

Reply:
> I'd like to get this filter in because some of the libraries (such as gutenberg-mobile) has a very costly setup (~20 min iirc) to get to a stage where we could publish the artifacts.

+1 for @oguzkocer's rationale. I found the exact same issue when working with Bintray. Hopefully a CI provider with more advanced configuration abilities than CircleCI will help tidying up this process. In the meantime, having it as a dedicated, centralized step makes the CI workflow clearer to look at.

Reply:
Thanks, I was lacking that context, makes more sense now that I have it with your explanation above.

Worth adding that feature request to the RFC p2 post about our future CI provider hunt then!!

Reply:
> Thanks, I was lacking that context, makes more sense now that I have it with your explanation above.

👍 

> Worth adding that feature request to the RFC p2 post about our future CI provider hunt then!!

For sure - I plan on posting a few things like this into discussion post. In the meantime, I've chatted about these with @jkmassel and it looks like our likely choice will address them all 🥳